sherpa/astro/ui/tests/test_astro_ui_background.py

```diff
@@ -1960,3 +1987,54 @@ def test_bkg_analysis_setting_changed(idval, direct, analysis, clean_astro_ui):
 
     assert src.units == analysis
     assert bkg.units == analysis
+
+
+@pytest.mark.parametrize("idval", [None, 1, "one"])
+def test_partially_set_bkg_models(idval, clean_astro_ui):
+    """We do not set a background model for all the backgrounds.
+
+    Regression test what happens.
+    """
+
+    ui.set_stat("leastsq")
+    ui.set_method("simplex")
+
+    idopt = 1 if idval is None else idval
+    ui.load_arrays(idopt, [1, 2, 3], [1, 1, 1], ui.DataPHA)
+    ui.get_data(idopt).name = f"fake-{idval}"
+
+    ui.set_bkg(idopt, DataPHA("up", [1, 2, 3], [0, 1, 1]), bkg_id="up")
+    ui.set_bkg(idopt, DataPHA("down", [1, 2, 3], [1, 0, 0]), bkg_id="down")
+
+    # Fake a response
+    #
+    egrid = np.asarray([0.2, 0.3, 0.4, 0.5])
+
+    ui.set_arf(idopt, create_arf(egrid[:-1], egrid[1:]))
+
+    # Set the source model
+    #
+    ui.set_source(idopt, ui.const1d.mdl)
+
+    # Ony set one of the background models
+    ui.set_bkg_source(idopt, ui.const1d.bmdldn, bkg_id="down")
+
+    emsg = f"^background model up for data set {idopt} has not been set$"
+    with pytest.raises(ModelErr, match=emsg):
+        ui.get_stat_info()
+
+    with pytest.raises(ModelErr, match=emsg):
+        ui.fit()
+
+    # Note that fit_bkg will skip the bkg_id=up data set.
+    #
+    ui.fit_bkg()
+    fres = ui.get_fit_results()
+    assert fres.datasets == (idopt, )
+    assert fres.succeeded
+    assert fres.parnames == ("bmdldn.c0", )
+    assert fres.numpoints == 3
+    assert fres.dof == 2
+    assert fres.istatval == pytest.approx(0.83)
+    assert fres.statval == pytest.approx(2 / 3)
+    assert fres.parvals == pytest.approx([3 + 1/3])
```

Comment:
I have trouble building sherpa again; happens every time I don't develop for a few weeks and every time I forget exactly what's wrong and every time it takes ma a while to get up and running again. Thus, I unfortunately won't try this out on my own right now, but I wonder why the fitted parameters is 3.3333, when the input data for the dependent axis in "down" os [1, 0, 0]. Should the best fit not be 0.33333 instead of 3.333333?

Reply:
Understood.

Reply:
There are places where I try to explain the values (because when I've reviewed them or when writing the tests) I want to explain it enough to be able to find again. Do you think it needed here (my suggestion is no, but YMMV)?